client/layout/masterbar/logged-in.jsx

```diff
@@ -201,140 +223,272 @@ class MasterbarLoggedIn extends Component {
 		);
 	}
 
-	render() {
-		const isWordPressActionSearchFeatureEnabled = config.isEnabled( 'wordpress-action-search' );
-		const {
-			domainOnlySite,
-			translate,
-			isCheckout,
-			isCheckoutPending,
-			isMigrationInProgress,
-			previousPath,
-			siteSlug,
-			isJetpackNotAtomic,
-			title,
-			currentSelectedSiteSlug,
-			currentSelectedSiteId,
-		} = this.props;
+	handleToggleMenu = () => {
+		this.setState( { isMenuOpen: ! this.state.isMenuOpen } );
+	};
 
-		const { isActionSearchVisible } = this.state;
+	dismissPopover = () => {
+		this.props.savePreference( MENU_POPOVER_PREFERENCE_KEY, true );
+	};
 
-		if ( isCheckout || isCheckoutPending ) {
-			return (
+	checkout = () => {
+		const { isCheckoutPending, previousPath, siteSlug, isJetpackNotAtomic, title } = this.props;
+		return (
+			<AsyncLoad
+				require="calypso/layout/masterbar/checkout.tsx"
+				placeholder={ null }
+				title={ title }
+				isJetpackNotAtomic={ isJetpackNotAtomic }
+				previousPath={ previousPath }
+				siteSlug={ siteSlug }
+				isLeavingAllowed={ ! isCheckoutPending }
+			/>
+		);
+	};
+
+	renderReader = () => {
+		const { translate } = this.props;
+		return (
+			<Item
+				tipTarget="reader"
+				className="masterbar__reader"
+				url="/read"
+				icon="reader"
+				onClick={ this.clickReader }
+				isActive={ this.isActive( 'reader' ) }
+				tooltip={ translate( 'Read the blogs and topics you follow' ) }
+				preloadSection={ this.preloadReader }
+			>
+				{ translate( 'Reader', { comment: 'Toolbar, must be shorter than ~12 chars' } ) }
+			</Item>
+		);
+	};
+
+	renderLanguageSwitcher = () => {
+		return (
+			( this.props.isSupportSession || config.isEnabled( 'quick-language-switcher' ) ) && (
```

Comment:
Same as above for early `return`-ing to make the code less nested and more readable.

I actually see this a few more times across the PR, so that comment applies for those other instances as well.